samples/AzureFunctionsApp/Entities/Counter.cs

```diff
@@ -0,0 +1,157 @@
+/// the added benefit of being able to use DI. When using TaskEntity<TState>, state is deserialized to the "State"
+/// property. No other properties on this type will be serialized/deserialized.
+/// </summary>
+public class Counter : TaskEntity<int>
+{
+    readonly ILogger logger;
+
+    public Counter(ILogger<Counter> logger)
+    {
+        this.logger = logger;
+    }
+
+    public int Add(int input)
+    {
+        this.logger.LogInformation("Adding {Input} to {State}", input, this.State);
+        return this.State += input;
+    }
+
+    public int OperationWithContext(int input, TaskEntityContext context)
+    {
+        // Get access to TaskEntityContext by adding it as a parameter. Can be with or without an input parameter.
+        // Order does not matter.
+        context.ScheduleNewOrchestration("SomeOrchestration", "SomeInput");
+
+        // When using TaskEntity<TState>, the TaskEntityContext can also be accessed via this.Context.
+        this.Context.ScheduleNewOrchestration("SomeOrchestration", "SomeInput");
+        return this.Add(input);
+    }
+
+    public int Get() => this.State;
+
+    [Function("Counter")]
+    public Task RunEntityAsync([EntityTrigger] TaskEntityDispatcher dispatcher)
+    {
+        // Can dispatch to a TaskEntity<TState> by passing a instance.
+        return dispatcher.DispatchAsync(this);
+    }
+
+    [Function("Counter_Alt")]
+    public static Task RunEntityStaticAsync([EntityTrigger] TaskEntityDispatcher dispatcher)
+    {
+        // Can also dispatch to a TaskEntity<TState> by using a static method.
+        return dispatcher.DispatchAsync<Counter>();
+    }
+
+    protected override int InitializeState(TaskEntityOperation operation)
+    {
+        // Optional method to override to customize initialization of state for a new instance.
+        return base.InitializeState(operation);
+    }
+}
+
+/// <summary>
+/// Example on how to dispatch to a POCO as the entity implementation. When using POCO, the entire object is serialized
+/// and deserialized.
+/// </summary>
+/// <remarks>
+/// Note there is a structural difference between <see cref="Counter"/> and <see cref="StateCounter"/>. In the former,
+/// the state is declared as <see cref="int"/>. In the later, state is the type itself (<see cref="StateCounter"/>).
+/// This means they have a different state serialization structure. The former is just "int", while the later is
+/// "{ \"Value\": int }".
+/// </remarks>
+public class StateCounter
+{
+    public int Value { get; set; }
+
+    public int Add(int input) => this.Value += input;
+
+    public int OperationWithContext(int input, TaskEntityContext context)
+    {
+        // Get access to TaskEntityContext by adding it as a parameter. Can be with or without an input parameter.
+        // Order does not matter.
+        context.ScheduleNewOrchestration("SomeOrchestration", "SomeInput");
```

Comment:
From what I can see, this operation is not actually called from anwhere, nor does it call an existing orchestration. It does serve as a syntax sample, but usually our samples demonstrate runnable scenarios. 

So it seems a bit out of place compared to the rest of the sample.

Reply:
Yeah, it was a syntax sample I threw together just to aid doc creation.. We can replace it with a runnable sample later.